atom-feed/src/main/java/org/openmrs/module/appointments/advice/AtomFeedAdviceActivatorComponent.java

```diff
@@ -0,0 +1,63 @@
+package org.openmrs.module.appointments.advice;
+
+import org.aopalliance.aop.Advice;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.annotation.OpenmrsProfile;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appointments.AppointmentsActivatorComponent;
+import org.openmrs.module.appointments.service.AppointmentRecurringPatternService;
+import org.openmrs.module.appointments.service.AppointmentServiceDefinitionService;
+import org.openmrs.module.appointments.service.AppointmentsService;
+
+/**
+ * Registers AOP advice when the context starts up.
+ * This is done instead of declaring advice in config.xml in order to allow it to be conditionally loaded
+ */
+@OpenmrsProfile(modules = { "openmrs-atomfeed:*" })
+public class AtomFeedAdviceActivatorComponent implements AppointmentsActivatorComponent {
+
+    private static final Log log = LogFactory.getLog(AtomFeedAdviceActivatorComponent.class);
+
+    private final AppointmentServiceDefinitionAdvice appointmentServiceDefinitionAdvice;
```

Comment:
I needed to use these properties and two different constructors just so that I could properly mock in unit tests.